PythonVisualizations/Array.py

```diff
@@ -278,8 +306,14 @@ def find(self, item):
 
     def find(self, val):
         self.startAnimations()
+        
+        # draw an index pointing to the last item in the list
+        indexDisplay = self.createIndex(len(self.list)-1, 'nItems', level = -1, color = 'black')
+        
         callEnviron = self.createCallEnvironment(

```

Comment:
There's no need to recreate the `nItems` index in each operation.  Use the one built in the `display()` method.